### PR TITLE
cap2.x gke implementation

### PR DESCRIPTION
**Pull request by prabalsharma**
- Bumps Helm version to 1.3.2
- Incorporates new folder structure, to have all the helper scripts and cap scripts in a common folder
- implements a retry for cert-issuer

### CODE REVIEW

File: cap-2.x/gke/main.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+module "cap-charts" {
+  source = "./common/cap-charts"
+
+  providers = {
+    helm = helm.helm-cap
+  }
+
+  cap_domain = "${module.gke-cluster.cluster_name}.${var.cap_domain}"
```
Comment (satadruroy) [inline cap-2.x/gke/main.tf:51]: Should be `cap_domain = var.cap_domain`

The UAA url doesn't have <cluster-name>.uaa.<domain> so the url above fails to connect from `wait_for_uaa.sh`

**Review by satadruroy — CHANGES_REQUESTED** [review]
Looks good except there's a minor bug for the UAA url.

File: cap-2.x/gke/main.tf (hunk 1 above)
Comment (prabalsharma) [reply cap-2.x/gke/main.tf]: done https://github.com/SUSE/cap-terraform/pull/97/commits/de575001c40a56e125ae03a793101678e089b42d

**Review by satadruroy — APPROVED** [review]

File: cap-2.x/aks/main.tf [hunk 2]
```diff
@@ -28,7 +28,7 @@ module "aks-cluster" {
 // kubeconfig will be generated by the AKS module
 
 provider "helm" {
-  version = "1.2.0"
+  version = "1.3.2"
```
Comment (bear454) [inline cap-2.x/aks/main.tf:31]: :+1:

File: cap-2.x/common/helper-charts/external-dns.tf [hunk 3]
```diff
@@ -1,17 +1,34 @@
 resource "helm_release" "external-dns" {
-  name  = "cap-external-dns"
-  repository = "https://marketplace.azurecr.io/helm/v1/repo"
```
Comment (bear454) [inline cap-2.x/common/helper-charts/external-dns.tf:3]: Bitnami explicitly recommends pulling from the Azure Marketplace when using with AKS:

https://bitnami.com/stack/external-dns/helm

File: cap-2.x/gke/main.tf (hunk 1 above)
Comment (satadruroy) [reply cap-2.x/gke/main.tf]: Clarification - the end-user would need to choose a unique subdomain, not the implementation (e.g. cluster name). Hence, no need for cluster name in front of the chosen domain.

File: cap-2.x/common/helper-charts/external-dns.tf (hunk 3 above)
Comment (satadruroy) [reply cap-2.x/common/helper-charts/external-dns.tf:3]: It's for efficiency as Azure Container Registry is geo-replicated so deployment on AKS would benefit from quicker image pulls. With our image sizes, this is a non-issue and we don't want to introduce conditional logic in the common chart for the registry url.

**Pull request merged**